### PR TITLE
Update User edit handlers

### PR DESCRIPTION
**Pull request by kburtram**
Fix most scenarios with the edit User dialog.

### CODE REVIEW

**Comment by kburtram** [comment]
Marking as draft while I finish a couple additional updates.

File: src/Microsoft.SqlTools.ServiceLayer/Security/UserActions.cs [hunk 1]
```diff
@@ -3,19 +3,477 @@
...
+            {
+                Success = true,
+                ErrorMessage = string.Empty
+            });
+        }
+
+        // /// <summary>
+        // /// Handle request to delete a user
+        // /// </summary>
+        // internal async Task HandleDeleteUserRequest(DeleteUserParams parameters, RequestContext<ResultStatus> requestContext)
```
Comment (kburtram) [inline src/Microsoft.SqlTools.ServiceLayer/Security/UserActions.cs:310]: dead code

**Review by caohai — APPROVED** [review]

**Comment by caohai** [comment]
I saw the added integration test failed at login code, taking a look

**Comment by kburtram** [comment]
@caohai thanks, I'll make that update.  I may push the integration tests separately since I'm validating this particular PR end-to-end manually.

**Pull request merged**